test/test_gauged.py

```diff
@@ -549,6 +549,78 @@ def test_clear_from(self):
         self.assertEqual(gauged.value('foo', timestamp=40000), 5)
         self.assertEqual(gauged.value('foo', timestamp=40000, namespace=1), 6)
 
+    def test_clear_key_before(self):
+        gauged = Gauged(self.driver, resolution=1000, block_size=10000)
+        with gauged.writer as writer:
+            writer.add({'foo': 1, 'bar': 1}, timestamp=10000)
+            writer.add({'foo': 2, 'bar': 2}, timestamp=20000)
+            writer.add({'foo': 3, 'bar': 3}, timestamp=30000)
+            writer.add({'foo': 4, 'bar': 4}, timestamp=40000, namespace=1)
+        self.assertEqual(gauged.value('foo', timestamp=40000), 3)
+        self.assertEqual(gauged.value('bar', timestamp=40000), 3)
+        self.assertEqual(gauged.value('foo', timestamp=40000, namespace=1), 4)
+        self.assertEqual(gauged.value('bar', timestamp=40000, namespace=1), 4)
+        with gauged.writer as writer:
+            writer.clear_key_before('foo', timestamp=20000)
+        # every value after 20000 stays the same
+        self.assertEqual(gauged.value('foo', timestamp=30000), 3)
+        self.assertEqual(gauged.value('bar', timestamp=30000), 3)
+        # 'foo' value before 20000 is cleared
+        self.assertEqual(gauged.value('foo', timestamp=10000), None)
```

Comment:
Can you add a test for `gauged.value('foo', timestamp=20000)` here to show if timestamps are inclusive?